pxtlib/cpp.ts

```diff
@@ -163,7 +163,17 @@ namespace pxt.cpp {
         let sourcePath = "/source/"
         let disabledDeps = ""
         let mainDeps: Package[] = []
-        for (let pkg of mainPkg.sortedDeps(true)) {
+
+        // order shouldn't matter for c++ compilation,
+        // so use a stable order to prevent order changes from fetching a new hex file
+        const mainPkgDeps = mainPkg.sortedDeps(true)
+            .sort((a, b) => {
+                if (a.id == "this") return 1;
+                else if (b.id == "this") return -1;
+                else return a.id.localeCompare(b.id);
```

Comment:
Use U.strcmp or at least compare. Locale may make it change based on language settings